### PR TITLE
kv: include conflicting request information in latch manager traces/logs

### LINKED ISSUE #114601: kv: include conflicting request information in latch manager traces/logs
The latch manager contains trace events and logs for conflicts in `Manager.waitForSignal`. This observability uses `latch.String` to present the span and the strength of the conflicting latch. However, it does not present information about the specific requests holding the latch or the requests acquiring the latch.

We should push down `*BatchRequest`s into the `spanlatch.Manager` and include the output from `BatchRequest.SafeFormat` in `latch.SafeFormat`.

Jira issue: CRDB-33589

Epic CRDB-34227

### PR DESCRIPTION
**Pull request by lyang24**
This commit logs latch waiter and latch holder's request information when latch
conflicts. This is achieved by adding batch request derived redact.SafeFormatter
into concurrency.Request and pass it down to latch Guard. Since the Guard struct
embedded the signal struct, we will store store a pointer to the latch guard
in latch struct instead.

Fixes: https://github.com/cockroachdb/cockroach/issues/114601

Relase note: None

### CODE REVIEW

**Review by lyang24 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten)

___
*[`pkg/kv/kvserver/concurrency/concurrency_control.go` line 438 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NkiP4mD2G36bR5G_9Mm:-NkiP4mD2G36bR5G_9Mn:bangpfl) ([raw file](https://github.com/cockroachdb/cockroach/blob/3bdb2d1d28855889aa919701cc180374c4602b28/pkg/kv/kvserver/concurrency/concurrency_control.go#L438)):*
> ```Go
> 	// We store a reference to kvpb.BatchRequest. This is used to enrich logging
> 	// when latches conflict.
> 	BatchRequest *kvpb.BatchRequest
> ```

probably not good to have duplicates (i.e. two mutable reference of transaction from req.Txn or req.BatchRequest.Txn). However, I think this is a safer change just in case BatchRequest changed downstream.
___
*[`pkg/kv/kvserver/spanlatch/manager.go` line 208 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NkiMcZQ0kWZb4e371kN:-NkiMcZQ0kWZb4e371kO:b-hih6gy) ([raw file](https://github.com/cockroachdb/cockroach/blob/3bdb2d1d28855889aa919701cc180374c4602b28/pkg/kv/kvserver/spanlatch/manager.go#L208)):*
> ```Go
> 				latch := &latches[i]
> 				latch.span = ss[i].Span
> 				latch.guard = guard
> ```

want to careful about potential npe here.
The old model is both latch and guard helds the same location to the signal in memory. In the new model this process become latch -> guard -> signal which requires guards lifetime >= latch's lifetime to avoid nil pointers - I think this statement is true because when latchGuard is releasing the latch also get deleted from the btree first?

**Review by lyang24 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten)

___
*[`pkg/kv/kvserver/concurrency/testdata/concurrency_manager/shared_locks_latches` line 172 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NlUnGaT35Q9RLp18ubh:-NlUnGaT35Q9RLp18ubi:bw600v9) ([raw file](https://github.com/cockroachdb/cockroach/blob/bca4dd05818405c1ff2bdf741a78a8fc6229765f/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/shared_locks_latches#L172)):*
> ```
> # exclusive_lock(ts) < shared_lock(ts)
> new-request name=req10 txn=txn3 ts=9,1
>   get key=a str=exclusive
> ```

get request with exclusive strength is interesting

**Comment by lyang24** [comment]
Hi @nvanbenschoten , I rebased this pr to the recent master that reflects the refactor of request safeformat. Do you mind take a look at this pr when you have time?

**Review by nvb — COMMENTED** [review]
> Reviewed 2 of 19 files at r1, 18 of 18 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @lyang24)

___
*[`pkg/kv/kvserver/concurrency/concurrency_control.go` line 438 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NkiP4mD2G36bR5G_9Mm:-NlyCeha69nEEqaNXdbe:b-g901uw) ([raw file](https://github.com/cockroachdb/cockroach/blob/3bdb2d1d28855889aa919701cc180374c4602b28/pkg/kv/kvserver/concurrency/concurrency_control.go#L438)):*
<details><summary><i>Previously, lyang24 (Lanqing Yang) wrote…</i></summary><blockquote>

probably not good to have duplicates (i.e. two mutable reference of transaction from req.Txn or req.BatchRequest.Txn). However, I think this is a safer change just in case BatchRequest changed downstream.
</blockquote></details>

I think this is ok as long as we change this to not be a direct reference to `BatchRequest`. Instead, we could make this a more narrow interface. For instance, it could just be `RequestFormatter redact.SafeFormatter`.

https://github.com/cockroachdb/cockroach/pull/116696 will be helpful for this.
___
*[`pkg/kv/kvserver/spanlatch/manager.go` line 208 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NkiMcZQ0kWZb4e371kN:-NlyFaFb1-9NsU8vVHhh:bnkavoi) ([raw file](https://github.com/cockroachdb/cockroach/blob/3bdb2d1d28855889aa919701cc180374c4602b28/pkg/kv/kvserver/spanlatch/manager.go#L208)):*
<details><summary><i>Previously, lyang24 (Lanqing Yang) wrote…</i></summary><blockquote>

want to careful about potential npe here.
The old model is both latch and guard helds the same location to the signal in memory. In the new model this process become latch -> guard -> signal which requires guards lifetime >= latch's lifetime to avoid nil pointers - I think this statement is true because when latchGuard is releasing the latch also get deleted from the btree first?
</blockquote></details>

`signals` is embedded in `Guard`, so the models are actually equivalent in that `latch` is pointing into memory in the `Guard`. The primary difference is that now it's pointing at the entire `Guard` struct instead of just field in it.
___
*[`pkg/kv/kvserver/spanlatch/manager.go` line 88 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NlyFzJD81irORTMu4zA:-NlyFzJD81irORTMu4zB:b-bc6zai) ([raw file](https://github.com/cockroachdb/cockroach/blob/bca4dd05818405c1ff2bdf741a78a8fc6229765f/pkg/kv/kvserver/spanlatch/manager.go#L88)):*
> ```Go
> // of a single key span.
> type latch struct {
> 	guard      *Guard
> ```

nit: let's just call this `g`.
___
*[`pkg/kv/kvserver/spanlatch/manager.go` line 108 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NlyGGTV15te3Di1ktc-:-NlyGGTW5c1xU2jtYdo7:bhoexuu) ([raw file](https://github.com/cockroachdb/cockroach/blob/bca4dd05818405c1ff2bdf741a78a8fc6229765f/pkg/kv/kvserver/spanlatch/manager.go#L108)):*
> ```Go
> 	w.Printf("%s@%s", la.span, la.ts)
> 	if la.guard != nil && la.guard.batchRequest != nil {
> 		w.Printf("[")
> ```

Instead of bracketing this, how about:
```go
w.Printf(" for request ")
la.guard.requestFormatter.SafeFormat(w, verb)
```
___
*[`pkg/kv/kvserver/concurrency/testdata/concurrency_manager/shared_locks_latches` line 172 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NlUnGaT35Q9RLp18ubh:-NlyH5NfDkHgXTm8jW8L:btk863o) ([raw file](https://github.com/cockroachdb/cockroach/blob/bca4dd05818405c1ff2bdf741a78a8fc6229765f/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/shared_locks_latches#L172)):*
<details><summary><i>Previously, lyang24 (Lanqing Yang) wrote…</i></summary><blockquote>

get request with exclusive strength is interesting
</blockquote></details>

It is! This is the request type that powers `SELECT FOR UPDATE`.
___
*[`pkg/kv/kvserver/spanlatch/manager_test.go` line 755 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NlyHSzi-djs6NVwRO6m:-NlyHSzi-djs6NVwRO6n:buwq6xe) ([raw file](https://github.com/cockroachdb/cockroach/blob/bca4dd05818405c1ff2bdf741a78a8fc6229765f/pkg/kv/kvserver/spanlatch/manager_test.go#L755)):*
> ```Go
> }
> 
> // Unit test the output of latch SafeFormat
> ```

Nice test!

nit: `// TestLatchStringAndSafeformat tests the output of latch.SafeFormat.`

**Review by lyang24 — COMMENTED** [review]
thanks for reviewing

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-:-NmA5xVW2FnVVOt5oLEv:b-p0kom7)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten)

___
*[`pkg/kv/kvserver/concurrency/concurrency_control.go` line 438 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NkiP4mD2G36bR5G_9Mm:-NmA5Ase-T3tqvRPQAqh:bo8plb6) ([raw file](https://github.com/cockroachdb/cockroach/blob/3bdb2d1d28855889aa919701cc180374c4602b28/pkg/kv/kvserver/concurrency/concurrency_control.go#L438)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

I think this is ok as long as we change this to not be a direct reference to `BatchRequest`. Instead, we could make this a more narrow interface. For instance, it could just be `RequestFormatter redact.SafeFormatter`.

https://github.com/cockroachdb/cockroach/pull/116696 will be helpful for this.
</blockquote></details>

Yes `redact.SafeFormatter` is a much better fit will wait for #116696  to merge and update
___
*[`pkg/kv/kvserver/spanlatch/manager.go` line 208 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NkiMcZQ0kWZb4e371kN:-NmA5Mvl2H7QtkvemUhL:bfrapvi) ([raw file](https://github.com/cockroachdb/cockroach/blob/3bdb2d1d28855889aa919701cc180374c4602b28/pkg/kv/kvserver/spanlatch/manager.go#L208)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

`signals` is embedded in `Guard`, so the models are actually equivalent in that `latch` is pointing into memory in the `Guard`. The primary difference is that now it's pointing at the entire `Guard` struct instead of just field in it.
</blockquote></details>

Great I learned about Struct Embedding in golang, very cool!

**Review by nvb — COMMENTED** [review]
:lgtm: other than the one remaining TODO. I'll try to get https://github.com/cockroachdb/cockroach/pull/116696 merged in the next day or two to unblock that.

> Reviewed 10 of 13 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-:-Nn9uwvX6-45D1Pi_c_L:b5es6ky)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @lyang24)

**Comment by lyang24** [comment]
seems like this had a small regression on benchmarks :(
```
./benchstat ~/Desktop/master.txt ~/Desktop/mybrach.txt 
goos: darwin
goarch: arm64
                                              │ /Users/lanqing/Desktop/master.txt │ /Users/lanqing/Desktop/mybrach.txt  │
                                              │              sec/op               │   sec/op     vs base                │
LatchManagerReadOnlyMix/size=1-12                                     130.5n ± 4%   144.9n ± 1%  +11.03% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=4-12                                     131.5n ± 2%   145.5n ± 1%  +10.68% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=16-12                                    133.3n ± 3%   143.9n ± 1%   +7.95% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=64-12                                    135.0n ± 2%   145.7n ± 2%   +7.97% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=128-12                                   136.4n ± 1%   144.4n ± 2%   +5.83% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=256-12                                   137.9n ± 1%   146.9n ± 1%   +6.53% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=0-12                           203.5n ± 0%   214.9n ± 3%   +5.63% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=1-12                           186.4n ± 2%   198.2n ± 2%   +6.30% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=4-12                           186.4n ± 2%   197.1n ± 1%   +5.74% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=16-12                          179.0n ± 4%   188.7n ± 1%   +5.39% (p=0.001 n=10)
LatchManagerReadWriteMix/readsPerWrite=64-12                          120.7n ± 4%   130.6n ± 1%   +8.21% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=128-12                         110.6n ± 4%   120.1n ± 2%   +8.50% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=256-12                         104.0n ± 5%   115.2n ± 2%  +10.77% (p=0.000 n=10)
geomean                                                               142.7n        153.7n        +7.71%

                                              │ /Users/lanqing/Desktop/master.txt │ /Users/lanqing/Desktop/mybrach.txt │
                                              │               B/op                │    B/op      vs base               │
LatchManagerReadOnlyMix/size=1-12                                      192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=4-12                                      192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=16-12                                     192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=64-12                                     192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=128-12                                    192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadOnlyMix/size=256-12                                    192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=0-12                            192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=1-12                            192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=4-12                            192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=16-12                           192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=64-12                           192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=128-12                          192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
LatchManagerReadWriteMix/readsPerWrite=256-12                          192.0 ± 0%    208.0 ± 0%  +8.33% (p=0.000 n=10)
geomean                                                                192.0         208.0       +8.33%

                                              │ /Users/lanqing/Desktop/master.txt │ /Users/lanqing/Desktop/mybrach.txt  │
                                              │             allocs/op             │ allocs/op   vs base                 │
LatchManagerReadOnlyMix/size=1-12                                      1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadOnlyMix/size=4-12                                      1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadOnlyMix/size=16-12                                     1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadOnlyMix/size=64-12                                     1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadOnlyMix/size=128-12                                    1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadOnlyMix/size=256-12                                    1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadWriteMix/readsPerWrite=0-12                            1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadWriteMix/readsPerWrite=1-12                            1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadWriteMix/readsPerWrite=4-12                            1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadWriteMix/readsPerWrite=16-12                           1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadWriteMix/readsPerWrite=64-12                           1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadWriteMix/readsPerWrite=128-12                          1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
LatchManagerReadWriteMix/readsPerWrite=256-12                          1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=10) ¹
geomean                                                                1.000        1.000       +0.00%
```

**Review by nvb — COMMENTED** [review]
Thanks for running these benchmarks! A regression of that magnitude is expected and acceptable, given the improvement this has on observability. We're adding 16 bytes (an interface header) to the `Guard` struct's size.

> Reviewed 1 of 13 files at r3, 8 of 9 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-:-NnLSuRPEPn5JeLpRPq0:b-o0k7vh)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @lyang24)

___
*[`pkg/kv/kvserver/concurrency/concurrency_control.go` line 438 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NkiP4mD2G36bR5G_9Mm:-NnLTG_HD36t6P6g4hN7:b2nm9wr) ([raw file](https://github.com/cockroachdb/cockroach/blob/3bdb2d1d28855889aa919701cc180374c4602b28/pkg/kv/kvserver/concurrency/concurrency_control.go#L438)):*
<details><summary><i>Previously, lyang24 (Lanqing Yang) wrote…</i></summary><blockquote>

Yes `redact.SafeFormatter` is a much better fit will wait for #116696  to merge and update
</blockquote></details>

This turned out nicely.
___
*[`pkg/kv/kvserver/concurrency/concurrency_control.go` line 437 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NnLS-5FCOXGjGtsMXUs:-NnLS-5FCOXGjGtsMXUt:b-rvzskh) ([raw file](https://github.com/cockroachdb/cockroach/blob/510bbad1c4f8d3cd19dcf216c6d0ee2d7e079c6b/pkg/kv/kvserver/concurrency/concurrency_control.go#L437)):*
> ```Go
> 	LockSpans *lockspanset.LockSpanSet
> 
> 	// The SafeFormatter from batchRequest pointer. This is used to enrich logging
> ```

"The SafeFormatter capable of formatting the request."
___
*[`pkg/kv/kvserver/concurrency/concurrency_control.go` line 439 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NnLSG8PA41lycDaYU-Z:-NnLSG8Q-sOg2mPTNaps:bx7gjiw) ([raw file](https://github.com/cockroachdb/cockroach/blob/510bbad1c4f8d3cd19dcf216c6d0ee2d7e079c6b/pkg/kv/kvserver/concurrency/concurrency_control.go#L439)):*
> ```Go
> 	// The SafeFormatter from batchRequest pointer. This is used to enrich logging
> 	// with request level information when latches conflict.
> 	BaFormatter redact.SafeFormatter
> ```

nit: consider switching the name of this field from `BaFormatter` to `BaFmt`. And then do the same in the functions where this is passed.
___
*[`pkg/kv/kvserver/spanlatch/manager.go` line 140 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NnLSV5n11xf1W_85FxQ:-NnLSV5n11xf1W_85FxR:bxg0aa6) ([raw file](https://github.com/cockroachdb/cockroach/blob/510bbad1c4f8d3cd19dcf216c6d0ee2d7e079c6b/pkg/kv/kvserver/spanlatch/manager.go#L140)):*
> ```Go
> 	// checking of conflicts, and waiting.
> 	snap        *snapshot
> 	baFormatter redact.SafeFormatter
> ```

nit: move this up to right below `pp`.
___
*[`pkg/kv/kvserver/concurrency/concurrency_manager_test.go` line 223 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NnLSjKs0jpDnmKObn__:-NnLSjKs0jpDnmKObn_a:b9vfzav) ([raw file](https://github.com/cockroachdb/cockroach/blob/510bbad1c4f8d3cd19dcf216c6d0ee2d7e079c6b/pkg/kv/kvserver/concurrency/concurrency_manager_test.go#L223)):*
> ```Go
> 					LockSpans:              lockSpans,
> 					PoisonPolicy:           pp,
> 					BaFormatter:            &ba,
> ```

small nit: let's just make `ba := &kvpb.BatchRequest{}` above.

**Review by lyang24 — COMMENTED** [review]
Thank you for reviewing!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-:-NnQSz-X38NJpBiArEWI:bzh25e2)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @nvanbenschoten)

___
*[`pkg/kv/kvserver/concurrency/concurrency_manager_test.go` line 223 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NnLSjKs0jpDnmKObn__:-NnQSn187aZ4XTL4gnuQ:b-30yzjx) ([raw file](https://github.com/cockroachdb/cockroach/blob/510bbad1c4f8d3cd19dcf216c6d0ee2d7e079c6b/pkg/kv/kvserver/concurrency/concurrency_manager_test.go#L223)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

small nit: let's just make `ba := &kvpb.BatchRequest{}` above.
</blockquote></details>

wow thanks for finding this, I applied to all the related test touched by this commit

**Review by nvb — COMMENTED** [review]
> Reviewed 5 of 9 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @lyang24)

___
*[`pkg/kv/kvserver/concurrency/concurrency_control.go` line 439 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NnLSG8PA41lycDaYU-Z:-NnQVhta3SQHSHtQkTM3:bbw93jv) ([raw file](https://github.com/cockroachdb/cockroach/blob/510bbad1c4f8d3cd19dcf216c6d0ee2d7e079c6b/pkg/kv/kvserver/concurrency/concurrency_control.go#L439)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

nit: consider switching the name of this field from `BaFormatter` to `BaFmt`. And then do the same in the functions where this is passed.
</blockquote></details>

Sorry, for parameters and unexported fields (in `Guard`), we'd want `baFmt`.
___
*[`pkg/kv/kvserver/spanlatch/manager_test.go` line 749 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NnQW7Bp-AjCAsO9_IZ9:-NnQW7Bp-AjCAsO9_IZA:b-ri1usz) ([raw file](https://github.com/cockroachdb/cockroach/blob/34fa3481be9e466537d3156791f965e6989258ae/pkg/kv/kvserver/spanlatch/manager_test.go#L749)):*
> ```Go
> 
> // TestSizeOfLatch tests the size of the latch struct.
> func TestSizeOfLatch(t *testing.T) {
> ```

While we're here, mind adding one of these tests for `Guard` as well?

**Review by lyang24 — COMMENTED** [review]
Thank you for speedy review i addressed the comments.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-:-NnQiJji621cr37MzgCU:bad7mbe)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @nvanbenschoten)

**Review by nvb — COMMENTED** [review]
> Reviewed 3 of 9 files at r7, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @lyang24)

___
*[`pkg/kv/kvserver/concurrency/concurrency_control.go` line 439 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NnLSG8PA41lycDaYU-Z:-NnQmGz43QVt2Laf5o1t:b-rtvrah) ([raw file](https://github.com/cockroachdb/cockroach/blob/510bbad1c4f8d3cd19dcf216c6d0ee2d7e079c6b/pkg/kv/kvserver/concurrency/concurrency_control.go#L439)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Sorry, for parameters and unexported fields (in `Guard`), we'd want `baFmt`.
</blockquote></details>

For function parameters as well (e.g. in `newGuard`). Variable names are rarely capitalized in Go.

**Review by lyang24 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @nvanbenschoten)

___
*[`pkg/kv/kvserver/concurrency/concurrency_control.go` line 439 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-NnLSG8PA41lycDaYU-Z:-NnQuFNq4ptexsWhpQce:b-na8l62) ([raw file](https://github.com/cockroachdb/cockroach/blob/510bbad1c4f8d3cd19dcf216c6d0ee2d7e079c6b/pkg/kv/kvserver/concurrency/concurrency_control.go#L439)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

For function parameters as well (e.g. in `newGuard`). Variable names are rarely capitalized in Go.
</blockquote></details>

my bad

**Review by nvb — APPROVED** [review]
:lgtm: Thanks @lyang24! I'll merge this once CI goes green.

> Reviewed 3 of 3 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115511#-:-NnR4iAf7MSRW5KbWYzF:b-scivxw)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @lyang24)

**Pull request merged**
